File: integration/sdks/protocol_smoke.mjs
import assert from "node:assert/strict";
import { Client } from "../../sdk/typescript/dist/index.js";

const [url, profile] = process.argv.slice(2);
const client = new Client(url, "sdk-synthetic-token");
try {
  // Concurrent cold calls must all wait for hello/epoch negotiation.
  await Promise.all(Array.from({ length: 8 }, () => client.call("ws.list")));
  const workspace = await client.createWorkspace({ name: "typescript-protocol-test" });
  try {
    assert.equal(workspace.spec.security.profile, profile);
    assert.equal((await client.getWorkspace(workspace.id)).id, workspace.id);
  } finally {
    await client.destroyWorkspace(workspace.id);
  }
} finally {
  await client.close();
}
console.log(`TypeScript strict-profile protocol passed: ${profile}`);

File: integration/sdks/protocol_smoke.py
"""Exercise the Python package against the real strict-profile Go relay/control."""

import asyncio
import sys

from remount import Client


async def main():
    url, profile = sys.argv[1:]
    async with Client(url, "sdk-synthetic-token") as client:
        assert client._controller_epoch > 0
        await asyncio.gather(*(client.call("ws.list") for _ in range(8)))
        workspace = await client.create_workspace({"name": "python-protocol-test"})
        try:
            assert workspace["spec"]["security"]["profile"] == profile
            assert (await client.get_workspace(workspace["id"]))["id"] == workspace["id"]
        finally:
            await client.destroy_workspace(workspace["id"])
    print(f"Python strict-profile protocol passed: {profile}")


asyncio.run(main())
